stingray/base.py

```diff
@@ -390,21 +689,702 @@ def write(self, filename: str, fmt: str = None) -> None:
+        object upon slicing.
+
+        This method adds functionality to retrieve the count value at
+        a particular index. This also can be used for slicing and generating
+        a new :class:`StingrayTimeseries` object. GTIs are recalculated based on the new light
+        curve segment
+
+        If the slice object is of kind ``start:stop:step``, GTIs are also sliced,
+        and rewritten as ``zip(time - self.dt /2, time + self.dt / 2)``
+
+        Parameters
+        ----------
+        index : int or slice instance
+            Index value of the time array or a slice object.
+
+        """
+        from .utils import assign_value_if_none
+
+        if isinstance(index, (int, np.integer)):
+            start = index
+            stop = index + 1
+            step = 1
+        elif isinstance(index, slice):
+            start = assign_value_if_none(index.start, 0)
+            stop = assign_value_if_none(index.stop, len(self))
+            step = assign_value_if_none(index.step, 1)
+        else:
+            raise IndexError("The index must be either an integer or a slice " "object !")
+
+        new_ts = type(self)()
+        for attr in self.meta_attrs():
+            setattr(new_ts, attr, copy.deepcopy(getattr(self, attr)))
+
+        for attr in self.array_attrs() + [self.main_array_attr]:
+            setattr(new_ts, attr, getattr(self, attr)[start:stop:step])
+
+        return new_ts
+
 
 class StingrayTimeseries(StingrayObject):
+    """Basic class for time series data.
+
+    This can be events, binned light curves, unevenly sampled light curves, etc. The only
+    requirement is that the data are associated with a time measurement.
+    We make a distinction between the *array* attributes, which have the same length of the
+    ``time`` array, and the *meta* attributes, which can be scalars or arrays of different
+    size. The array attributes can be multidimensional (e.g. a spectrum for each time bin),
+    but their first dimension (``array.shape[0]``) must have same length of the ``time`` array.
+
+    Array attributes are singled out automatically depending on their shape. All filtering
+    operations (e.g. ``apply_gtis``, ``rebin``, etc.) are applied to array attributes only.
+    For this reason, it is advisable to specify whether a given attribute should *not* be
+    considered as an array attribute by adding it to the ``not_array_attr`` list.
+
+    Parameters
+    ----------
+    time: iterable
+        A list or array of time stamps
+
+    Other Parameters
+    ----------------
+    array_attrs : dict
+        Array attributes to be set (e.g. ``{"flux": flux_array, "flux_err": flux_err_array}``).
+        In principle, they could be specified as simple keyword arguments. But this way, we
+        will run a check on the length of the arrays, and raise an error if they are not of a
+        shape compatible with the ``time`` array.
+
+    dt: float
```

Comment:
It's in the "other parameters", absolutely optional, and it can be 0 or None for events. Also, it can be a single float for light curves, if uniformly sampled.
But it _might_ be useful if we are mixing up events with different frame times, for example.